tests/integration_tests/flows/test_clickhouse.py

```diff
@@ -44,92 +56,226 @@ def query_ch(query, database='default'):
     )
 
     if res.status_code != 200:
-        print(f'Error in query: {query}')
+        print(f'ERROR: code={res.status_code} msg={res.text}')
+        raise Exception()
 
     if ' FORMAT JSON' in query:
-        return res.json()['data']
-    else:
-        return res.text
+        res = res.json()['data']
+
+    return res
+
 
 class ClickhouseTest(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        set_get_config_path()
-
-        cls.sp = Popen(['python3', '-m', 'mindsdb'], close_fds=True)
-
-        for i in range(20):
-            try:
-                res = requests.get(f'{root}/util/ping')
-                if res.status_code != 200:
-                    raise Exception('')
-            except:
-                time.sleep(1)
-                if i == 19:
-                    raise Exception("Can't connect !")
-
-        #query_ch('DROP DATABASE mindsdb')
-        query_ch(f"""
-        CREATE TABLE {ds_name} (number_of_rooms String, number_of_bathrooms String, sqft Int64, location String, days_on_market Int64, initial_price Int64, neighborhood String, rental_price Float64)  ENGINE=URL('https://raw.githubusercontent.com/mindsdb/mindsdb-examples/master/benchmarks/home_rentals/dataset/train.csv', CSVWithNames)
+        cls.mdb = MindsdbNative(config)
+
+        models = cls.mdb.get_models()
+        models = [x['name'] for x in models]
+        if TEST_PREDICTOR_NAME in models:
+            cls.mdb.delete_model(TEST_PREDICTOR_NAME)
```

Comment:
can be replaced with `MindsdbNative(config).delete_model(TEST_PREDICTOR_NAME)`

Reply:
.delete_model raise error if predictor not exists, so need check it before try to delete

Reply:
fair enough, still no point in attaching the `mdb` object to `cls`.

Reply:
mdb object used in tests to check what test predictor really created/removed